File: JS/SignOfTheProductOfAnArray.js
/**
 * @version JAVASCRIPT ECMAScript 6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
function arraySign(nums) {
    if (nums.includes(0)) return 0;
    
    let minusTimes = 0;
    nums.forEach((item) => {
        if (item < 0) {
            minusTimes++;
        }
    });
    
    return minusTimes % 2 === 0 ? 1 : -1;
}
export { arraySign }

File: GO/SignOfTheProductOfAnArray.go
/**
 * @version GO 1.20.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func arraySign(nums []int) int {
	var signFunc = func(num int) int {
		if num > 0 {
			return 1
		} else if num < 0 {
			return -1
		} else {
			return 0
		}
	}

	res := 1
	for _, num := range nums {
		res *= signFunc(num)
	}

	return res
}
